Makefile

```diff
@@ -1,43 +1,84 @@
 all: deps
 
+# List all commands
+.PHONY: ls
+ls:
+	@$(MAKE) -pRrq -f $(lastword $(MAKEFILE_LIST)) : 2>/dev/null | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($$1 !~ "^[#.]") {print $$1}}' | sort | egrep -v -e '^[^[:alnum:]]' -e '^$@$$' | xargs
+
 ###############################
 #    Project-wide commands    #
 ###############################
 
 .PHONY: deps
 deps:
-	(cd ./web ; yarn install)
-	(cd ./serverless ; yarn install)
+	@echo Installing web app dependencies...
+	@(cd ./web ; yarn install)
+	@echo Installing functions dependencies...
+	@(cd ./functions ; yarn install)
 
 .PHONY: test
 test:
-	(cd ./serverless ; yarn test)
-	(cd ./web ; yarn test)
+	@echo Running web app tests...
+	@(cd ./web ; yarn test)
+	@echo Running function tests...
+	@(cd ./functions ; yarn test)
 
 .PHONY: lint
 lint:
-	(cd ./serverless ; yarn lint)
-	(cd ./web ; yarn lint)
+	@echo Linting web app...
+	@(cd ./web ; yarn lint)
+	@echo Linting functions...
+	@(cd ./functions ; yarn lint)
 
 .PHONY: report-coverage
 report-coverage:
 	mkdir -p .nyc_output
-	cp serverless/coverage/coverage-final.json .nyc_output/coverage-serverless.json
+	cp functions/coverage/coverage-final.json .nyc_output/coverage-functions.json
 	cp web/coverage/coverage-final.json .nyc_output/coverage-web.json
 	nyc report --reporter=text-lcov | coveralls
 
 ###############################
 # Component-specific commands #
 ###############################
 
+# Builds front-end
 .PHONY: build
 build:
-	(cd ./web ; yarn build)
+	@echo Building web app...
+	@(cd ./web ; yarn build)
 
+# Runs webpack dev server
 .PHONY: web
 web:
-	(cd ./web ; yarn start)
+	@echo Starting webpack-dev-server...
+	@(cd ./web ; yarn start)
+
+# Runs a simple web serve to test the build static web app
+.PHONY: serve
+serve:
+	@echo Go to http://localhost:8081 to test the built web app
+	@(cd ./docs ; python ../serve.py)
```

Comment:
Can we use [http-server](https://www.npmjs.com/package/http-server) instead of the Python script? I know Python 2 & 3 come installed by default on Macs but that's not the case for Windows.  After running `yarn add http-server -D` in web, we can replace line 60 in the Makefile with `@(./web/node_modules/.bin/http-server ./docs -a localhost -p 8081 -c-1)`.

Reply:
That would be nice because requiring the user to have Python 2 installed is more of an inconvenience than installing an additional npm package in my mind. Also avoids having to create a new file just for serving the app. Is there a reason why we wouldn't do mock data filling in JavaScript? Imo it would be really nice to stick to one language unless we have a compelling reason not to.

Reply:
No reason, one language is a bit boring and I just find python more convenient, and python2 is here because as you mentioned its installed by default on osx. But javascript is fine too.

Fair point on the extra file (though extra npm dependency 😢 ) - i'll make the changes tonight to use the npm package instead

Reply:
542a586